projects/opendr_ws_2/src/opendr_perception/opendr_perception/panoptic_segmentation_efficient_ps.py

```diff
@@ -0,0 +1,191 @@
+            # Retrieve a list of two OpenDR heatmaps: [instance map, semantic map]
+            prediction = self._learner.infer(image)
+
+            # The output topics are only published if there is at least one subscriber
+            if self._visualization_publisher is not None and self._visualization_publisher.get_subscription_count() > 0:
+                panoptic_image = EfficientPsLearner.visualize(image, prediction, show_figure=False,
+                                                              detailed=self.detailed_visualization)
+                self._visualization_publisher.publish(self._bridge.to_ros_image(panoptic_image))
+
+            if self._instance_heatmap_publisher is not None and self._instance_heatmap_publisher.get_subscription_count() > 0:
+                self._instance_heatmap_publisher.publish(self._bridge.to_ros_image(prediction[0]))
+            if self._semantic_heatmap_publisher is not None and self._semantic_heatmap_publisher.get_subscription_count() > 0:
+                self._semantic_heatmap_publisher.publish(self._bridge.to_ros_image(prediction[1]))
+
+        except Exception as e:
+            self.get_logger().error(f'Failed to generate prediction: {e}')
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('input_rgb_image_topic', type=str, default='/usb_cam/image_raw',
+                        help='listen to RGB images on this topic')
+    parser.add_argument('--checkpoint', type=str, default='cityscapes',
+                        help='download pretrained models [cityscapes, kitti] or load from the provided path')
+    parser.add_argument('--output_heatmap_topic', type=str, default='/opendr/panoptic',
+                        help='publish the semantic and instance maps on this topic as "OUTPUT_HEATMAP_TOPIC/semantic" \
+                             and "OUTPUT_HEATMAP_TOPIC/instance"')
+    parser.add_argument('--output_rgb_image_topic', type=str,
+                        default='/opendr/panoptic/rgb_visualization',
+                        help='publish the panoptic segmentation map as an RGB image on this topic or a more detailed \
+                              overview if using the --detailed_visualization flag')
+    parser.add_argument('--detailed_visualization', action='store_true',
+                        help='generate a combined overview of the input RGB image and the semantic, instance, and \
+                              panoptic segmentation maps and publish it on OUTPUT_RGB_IMAGE_TOPIC')
+    args = parser.parse_args()
+
+    efficient_ps_node = EfficientPsNode(args.input_rgb_image_topic,
+                                        args.checkpoint,
+                                        args.output_heatmap_topic,
+                                        args.output_rgb_image_topic,
+                                        args.detailed_visualization)
+    efficient_ps_node.listen()
```

Comment:
Regarding this section of the node, there are multiple issues that stop the node from running:

1. `rclpy.init()` needs to be called before creating the object from the node class and not afterwards inside the `listen()` method.

2. The general syntax of main is causing an error: `AttributeError: module 'opendr_perception.panoptic_segmentation_efficient_ps' has no attribute 'main'`

    This is a weird issue that can be resolved by using syntax similar [to other nodes](https://github.com/opendr-eu/opendr/blob/4be61a9e06d53701c3928b3c4219aac29d128409/projects/opendr_ws_2/src/opendr_perception/opendr_perception/pose_estimation_node.py#L111-L169) with a `main()` function defined.

3. Regarding the arguments, the input image argument now uses a default value but is declared as a required argument instead of optional (with `--`). Moreover, it would be helpful to define short options for arguments similar to other nodes.  

    The suggested short options are consistent with the [semantic segmentation node arguments](https://github.com/opendr-eu/opendr/blob/4be61a9e06d53701c3928b3c4219aac29d128409/projects/opendr_ws_2/src/opendr_perception/opendr_perception/semantic_segmentation_bisenet_node.py#L154-L165) that has similar arguments and output.

I include fixes for all three issues in the suggestion below in conjunction with the suggestion in the listen method:

```suggestion
def main(args=None):
    rclpy.init(args=args)
    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument('-i', '--input_rgb_image_topic', type=str, default='/usb_cam/image_raw',
                        help='listen to RGB images on this topic')
    parser.add_argument('--checkpoint', type=str, default='cityscapes',
                        help='download pretrained models [cityscapes, kitti] or load from the provided path')
    parser.add_argument('-o', '--output_heatmap_topic', type=str, default='/opendr/panoptic',
                        help='publish the semantic and instance maps on this topic as "OUTPUT_HEATMAP_TOPIC/semantic" \
                                 and "OUTPUT_HEATMAP_TOPIC/instance"')
    parser.add_argument('-v', '--output_rgb_image_topic', type=str,
                        default='/opendr/panoptic/rgb_visualization',
                        help='publish the panoptic segmentation map as an RGB image on this topic or a more detailed \
                                  overview if using the --detailed_visualization flag')
    parser.add_argument('--detailed_visualization', action='store_true',
                        help='generate a combined overview of the input RGB image and the semantic, instance, and \
                                  panoptic segmentation maps and publish it on OUTPUT_RGB_IMAGE_TOPIC')
    args = parser.parse_args()

    efficient_ps_node = EfficientPsNode(args.input_rgb_image_topic,
                                        args.checkpoint,
                                        args.output_heatmap_topic,
                                        args.output_rgb_image_topic,
                                        args.detailed_visualization)
    efficient_ps_node.listen()


if __name__ == '__main__':
    main()
```

Please verify that the node is running properly after applying the fixes.